pkg/api/metrics.go

```diff
@@ -17,36 +18,120 @@ import (
 	"github.com/grafana/grafana/pkg/tsdb/legacydata"
 )
 
+var (
+	ErrNoQueries                   = errors.New("no queries found")
+	ErrMissingDatasourceIdentifier = errors.New("missing data source ID/UID")
+	ErrDifferentDatasources        = errors.New("all queries must use the same datasource")
+)
+
+type MissingDatasourceInfoError struct {
+	RefID string
+}
+
+func (m MissingDatasourceInfoError) Error() string {
+	return fmt.Sprintf("query mising datasource info: %s", m.RefID)
+}
+
+func (hs *HTTPServer) handleQueryMetricsError(err error) *response.NormalResponse {
+	hs.log.Debug("Error while querying data", "err", err)
```

Comment:
Not needed since the error provided in `response.Error` will be logged as an error.

```suggestion
```